### DISCUSSION TITLE
Which versions are active, which are deprecated?

### DISCUSSION THREAD

**Discussion by bquorning** [jruby/jruby#7915] [Q&A] [answered]
When deciding whether to drop Ruby 2.6 compatibility from some private gems, we realised that that would also remove their compatibility with JRuby 9.3.

This led us to ask: which versions of JRuby are maintained, and which are deprecated? We found https://github.com/jruby/jruby/wiki/Roadmap and https://github.com/jruby/jruby/blob/master/SECURITY.md which contain relevant, but outdated information.

**Comment by ahorek** [comment]
yes, these articles you've mentioned are a bit outdated.

9.4.x currently targets Ruby 3.1.4 support.
9.3.x is Ruby 2.6.x support level (if you need compatibility), it's still maintained, but probably not for long.
9.2.x (Ruby 2.5.x) is no longer supported

**Comment by bquorning** [comment]
Thanks for the response @ahorek. I am curious if you have more details about the “but probably not for long” part regarding JRuby 9.3.x.

**Comment by ahorek** [comment]
note that Ruby 2.6 is already EOL for a year now.

it takes some to for JRuby to catch up, so the maintenance lasts longer, but usually, 2 branches are actively maintained. At the moment, it's 9.4.x and 9.3.x. The main focus is of course always the latest version, but some features or bugfixes could still be backported to 9.3.

anyway, the exact support timeline is a question for maintainers.

the wiki should be updated accordingly, that's a good point.

**Comment by headius** [comment, marked as answer]
Sorry we have not made this clear! @ahorek is correct in the general details; 9.4 is currently developed, 9.3 is in maintenance, and 9.2 and lower are EOL. 9.3 will be supported at least until the end of this year. JRuby 9.5 will hopefully be coming this winter, at which point we would "soft EOL" 9.3 to encourage people to catch up.

We don't make exact time commitments for support because different release levels see more or less adoption, and we don't want to abandon users unnecessarily.

**Comment by headius** [comment]
I'm converting this to a discussion since there's not really anything to fix here.
